proxy/src/telemetry/sensor/http.rs

```diff
@@ -54,15 +55,25 @@ struct RespondInner {
     request_open: Instant,
 }
 
-#[derive(Default, Debug)]
-pub struct ResponseBody<B> {
+pub type ResponseBody<B> = MeasuredBody<B, ResponseBodyInner>;
+pub type RequestBody<B> = MeasuredBody<B, RequestBodyInner>;
+
+#[derive(Debug)]
+pub struct MeasuredBody<B, I: BodySensor> {
     body: B,
-    inner: Option<ResponseBodyInner>,
+    inner: Option<I>,
     _p: PhantomData<(B)>,
 }
 
+pub trait BodySensor: Sized {
```

Comment:
I was confused at first why there was a new trait, but I understood once I read the impls. Perhaps a small doc comment here could say why this trait exists?

Reply:
Sure, will do!